src/monitoring_service/service.py

```diff
@@ -50,12 +51,25 @@ def check_gas_reserve(web3: Web3, private_key: str) -> None:
 
 
 def handle_event(event: Event, context: Context) -> None:
+    """ Calls the handler for the given event.
+
+    Exceptions are caught and generate both error logs and sentry issues.
+    Events are not retried after an exception.
+    """
     log.debug("Processing event", event_=event)
     handler = HANDLERS.get(type(event))
 
     if handler:
-        handler(event, context)
-        log.debug("Processed event", num_scheduled_events=context.db.scheduled_event_count())
+        with sentry_sdk.push_scope() as sentry_scope:
+            sentry_scope.set_tag("event", event.__class__.__name__)
+            try:
+                handler(event, context)
+                log.debug(
+                    "Processed event", num_scheduled_events=context.db.scheduled_event_count()
+                )
+            except Exception as ex:  # pylint: disable=broad-except
+                log.error("Error during event handler", event=event, exc_info=ex)
+                sentry_sdk.capture_exception(ex)
```

Comment:
We need to check whether or not this works when sentry hasn't been initialized.

Reply:
I did a very basic tests and apparently it does:
```
Python 3.7.1 (default, Nov 12 2018, 13:22:25) 
[GCC 6.3.0 20170516] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sentry_sdk
>>> sentry_sdk.capture_exception(Exception('foo'))
>>> 
```